app/src/main/java/com/example/movieapp/ui/MainViewModel.kt

```diff
@@ -0,0 +1,54 @@
+package com.example.movieapp.ui
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.movieapp.data.MovieRepository
+import com.example.movieapp.models.MovieMapper
+import com.example.movieapp.models.MovieUi
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+class MainViewModel(
+    private val movieRepository: MovieRepository
+) : ViewModel() {
+
+    private val _stateFlow = MutableStateFlow(emptyList<MovieUi>())
+    val stateFlow: StateFlow<List<MovieUi>> get() = _stateFlow.asStateFlow()
+
+    private val totalPageCount = 20
+    private var currentPage = 1
+    val isLastPage = currentPage == totalPageCount
+    var isLoading = false
+
+    init {
+        loadMovies()
+    }
+
+    private fun loadMovies() {
+        viewModelScope.launch(Dispatchers.IO) {
+            isLoading = true
+
+            val list = movieRepository.loadPopularMovies(currentPage).map { movieDto ->
+                MovieMapper.map(movieDto)
+            }
+            _stateFlow.emit(list)
```

Comment:
you should use `_state.update {} ` for updating the state of a `MutableStateFlow`

_state: MutableStateFlow = MutableStateFlow(123)
state = _stateFlow.asStateFlow()


// to update
_state.update {
    // new value
    151
}

data class Person(
    val name: String = "Luka",
    val lastName: String = "Lukovic"
)
class s {
    
    
    private val _state = MutableStateFlow(Person())
    private val state = _state.asStateFlow()
    
    init {
        _state.update { person -> 
            person.copy(
                name = "Ana",
                lastName = "Bolovic"
            )
        }
    }
}

Reply:
Done